### PR TITLE
Add rholang-tutorial to main repo

### PR DESCRIPTION
**Pull request by tschoffelen**
## Overview
As requested by @KellyatPyrofex, this PR merges the [tschoffelen/rholang-tutorial](https://github.com/tschoffelen/rholang-tutorial) repo into the subdirectory `rholang-tutorial`.

### Does this PR relate to an RChain JIRA issue? 
N/A

### Complete this checklist before you submit the PR
- [x] This PR contains no more than 200 lines of code, excluding test code.
- [x] This PR meets [RChain development coding standards](https://rchain.atlassian.net/wiki/spaces/DOC/pages/28082177/Coding+Standards).
- [x] You have someone in mind to assign for review. Make this assignment after submitting the PR.

### Notes
N/A

### CODE REVIEW

File: rholang-tutorial/yarn.lock [hunk 1]
```diff
@@ -0,0 +1,8682 @@
+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
```
Comment (pyrocto) [inline rholang-tutorial/yarn.lock]: Does this need to be checked in, or can it be generated with yarn build?

File: rholang-tutorial/src/components/Markdown.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+      if(!setToc) {
+        toc.push({type: node.name, id: props.id, value: flattenedChildren})
+      }
+
+      return React.createElement(Component, props, children)
+    }
+
+    // Transform any code examples
+    if (node.type === 'tag' && node.name === 'pre') {
```
Comment (pyrocto) [inline rholang-tutorial/src/components/Markdown.js]: Only code samples with line numbers should be converted, so include a check that /^\s*\d+/ matches the first line.

File: rholang-tutorial/yarn.lock (hunk 1 above)
Comment (tschoffelen) [reply rholang-tutorial/yarn.lock]: Yes, this is a file that locks in certain versions of dependencies. From the [Yarn docs](https://yarnpkg.com/lang/en/docs/yarn-lock/#toc-check-into-source-control):

> All yarn.lock files should be checked into source control (e.g. git or mercurial). This allows Yarn to install the same exact dependency tree across all machines, whether it be your coworker’s laptop or a CI server.

File: rholang-tutorial/src/components/Markdown.js (hunk 2 above)
Comment (tschoffelen) [reply rholang-tutorial/src/components/Markdown.js]: Actually, I think all code samples should be converted from boring `<pre>`'s to their syntax highlighted counterparts.

There is already code present to check whether a 'Run' button should be shown depending on whether there are any line numbers in the sample: https://github.com/rchain/rchain/pull/683/files#diff-a59c22e3ffbd92939dc8f2d14b61c6c1R19

(related issue from earlier today: https://github.com/tschoffelen/rholang-tutorial/issues/1)

Comment (tschoffelen) [reply rholang-tutorial/src/components/Markdown.js]: Thanks for checking this so thoroughly by the way! @pyrocto

Comment (pyrocto) [reply rholang-tutorial/src/components/Markdown.js]: Sweet! LGTM.

**Review by ghost — APPROVED** [review]

**Comment by azazime** [comment]
Nice work

**Pull request merged**